clients/fides-js/src/fides-tcf.ts

```diff
@@ -235,15 +246,25 @@ const updateFidesCookieFromString = (
 const init = async (config: FidesConfig) => {
   const overrideOptions: Partial<FidesOptionOverrides> =
     getOverrideFidesOptions();
+  const customPrefs: GetPreferencesFnResp | null = await getConsentPreferences(
```

Comment:
again, thinking of streamlining our initialization logic...

these are technically override options, aren't they? can we call this in `getOverrideFidesOptions`? it's hard to follow when these `customPrefs` are being applied in 3 different places in this function. 

in other words, is there a way to call `getConsentPreferences` in `getOverrideFidesOptions` and override all the fields we need in there before the rest of the function runs?

Reply:
I haven't included them in `getOverrideFidesOptions` because a `getConsentPreferences` fn can only be provided in the `init` and not through all of the following: window obj, cookie, query param. So they're a bit different. 

> is there a way to call getConsentPreferences in getOverrideFidesOptions and override all the fields we need in there before the rest of the function runs?

I actually tried this route initially, but since we obtain the values we need to override later in the init lifecycle (e.g. `cookie.tcf_version_hash`, we have to pass down the return vals from `getConsentPreferences`, and cannot override all fields immediately

Reply:
hmm okay, thinking out loud here... so there are 3 fields at play here:
### `consent`
This needs to override part of `cookie`. It's been a while since I looked at this code, but I think if the `config` object has a `consent` object, `getInitialCookie` will just use that (through `makeConsentDefaultsLegacy`)? So instead of 

```js
if (customPrefs?.consent) {
    cookie.consent = customPrefs.consent;
}
```
I think `getOverrideFidesOptions` could also overwrite `cookie.consent` and that'll propagate down okay. I know the override func only has `Partial<FidesOptionOverrides>` right now though... if you're against changing that, I feel like we could still maybe take advantage of the `getInitialCookie` being willing to take a `consent` obj.

### `fides_string`
This can fill in `overrideOptions.fidesString` (and indeed does so right on the next line) so it feels reasonable to me to put it with `getOverrideFidesOptions` 

### `version_hash`
I think `tcf_version_hash` can be written whenever—the only time we write to it is when we save an experience. so there's no other part in the initialization logic that that writes to `tcf_version_hash`

Reply:
So it ended up being easier than I thought to go ahead and update `getOverrideFidesOptions = async(config: FidesConfig): Promise<Partial<FidesOptionOverrides>>` to include all overrides, like:  `getOverrides = async(config: FidesConfig): Promise<Partial<FidesOverrides>>

Thanks for the suggestions!